src/daft-dsl/src/arithmetic.rs

```diff
@@ -43,8 +32,8 @@ mod tests {
     fn check_add_expr_type() -> DaftResult<()> {
         let a = col("a");
         let b = col("b");
-        let c = a + b;
-        match c {
+        let c = a.add(b);
```

Comment:
I actually quite liked being able to use the Rust arithmetic operators on logical expressions in Rust unit tests, since it made things quite readable.

Couldn't we implement the `std::ops::*` traits on `ExprRef` (i.e. `Arc<Expr>`)? This should allow use to keep using these arithmetic operators without the extra cloning, since the trait implementation should match our current `ExprRef`-oriented signature.